### PR TITLE
Auth kit update

### PR DESCRIPTION
**Pull request by ademidun**
Create a more comprehensive introduction to the Auth Kit documentation including a video demo, more complete and copy-paste-friendly code snippets, and troubleshooting tips.

Verified this works in https://github.com/5afe/safe-space/pull/12

https://user-images.githubusercontent.com/9806858/234088191-441c1c18-5922-47a4-a4dd-44fb383a54d6.mp4

### CODE REVIEW

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 1]
```diff
@@ -4,51 +4,104 @@ The [Auth kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/a
...
-const web3AuthAdapter = new Web3AuthAdapter(options, [openloginAdapter], modalConfig);
+## Sign In to an Ethereum Account
 
-// Create an instance of the SafeAuthKit using the adapter and the SafeAuthConfig allowed options
-const safeAuthKit = await SafeAuthKit.init(web3AuthAdapter, { ... });
-```
+Once your Auth Kit instance is created, call `signIn()` to start the authentication process.
 
-Once the instance is created, you can call the `signIn()` method to start the authentication process showing the web3Auth modal in case you use the `Web3AuthAdapter`.
+When you sign in with the same social account, the same Ethereum address will be returned.
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:102]: When you sign in with the same social account, the same Ethereum address will be returned for the same client id.

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 2]
```diff
@@ -92,9 +145,42 @@ const safeAuthKit = await SafeAuthKit.init(web3AuthAdapter, {
...
+
+```typescript
+import { ethers } from 'ethers'
+import { EthersAdapter } from '@safe-global/protocol-kit'
+
+provider = new ethers.providers.Web3Provider(safeAuthKit.getProvider()!);
+signer = provider.getSigner();
+
+const ethAdapter = new EthersAdapter({
+      ethers,
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:162]: The usual spacing is 2 in our repos

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 3]
```diff
@@ -4,51 +4,104 @@ The [Auth kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/a
...
+- [Create a Web3Auth account](https://web3auth.io)
+
+## Example
+
+See an [Auth Kit example in the Safe Core SDK](https://github.com/safe-global/safe-core-sdk/blob/7959821ab08c96cf3babb9ed906c01d644ac49f4/packages/auth-kit/example/src/App.tsx#L33) and [Pull Request 12](https://github.com/5afe/safe-space/pull/12) in [Safe Space](https://github.com/5afe/safe-space).
+
+
+{% embed url="https://user-images.githubusercontent.com/9806858/234088191-441c1c18-5922-47a4-a4dd-44fb383a54d6.mp4" %}
+[Watch video demo](https://user-images.githubusercontent.com/9806858/234088191-441c1c18-5922-47a4-a4dd-44fb383a54d6.mp4)
+{% endembed %}
```
Comment (dasanra) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:23]: I think there is a misspell in this tag

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 4]
```diff
@@ -115,6 +201,72 @@ await signer.signTransaction(tx);
 await signer.signMessage(message);
 ```
 
-## Example
+## Fix Webpack 5 Polyfills Issue
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:204]: Hey @ademidun I think would be enough with a link to this known problem to any website helping with this. Is a very long section unrelated to the kit itself 😬

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md (hunk 3 above)
Comment (ademidun) [reply learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:23]: Which one is misspelled? `endembed`? That is a Gitbook code for embedding videos: https://github.com/safe-global/safe-docs/blob/main/learn/safe-core/safe-core-account-abstraction-sdk/onramp-kit.md?plain=1#L19

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md (hunk 1 above)
Comment (ademidun) [reply learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:102]: Thanks! Good catch! Important distinction.
```suggestion
When you sign in with the same social account, the same Ethereum address will be returned for the same Client ID.
```

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md (hunk 2 above)
Comment (ademidun) [reply learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md]: Done: f47c33b

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md (hunk 4 above)
Comment (ademidun) [reply learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md]: Done: f47c33b

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 5]
```diff
@@ -4,51 +4,123 @@ The [Auth kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/a
...
+  },
+  adapterSettings: {
+    uxMode: 'popup',
+    whiteLabel: {
+      name: 'Safe'
+    }
+  }
+})
+
+const adapter = new Web3AuthAdapter(options, [openloginAdapter], modalConfig)
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:110]:
```suggestion
const web3AuthModalPack = new Web3AuthModalPack(options, [openloginAdapter], modalConfig)
```

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 6]
```diff
@@ -4,51 +4,123 @@ The [Auth kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/a
...
+
+const safeAuthKit = await SafeAuthKit.init(adapter, {
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:112]:
```suggestion
const safeAuthKit = await SafeAuthKit.init(web3AuthModalPack, {
```

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 7]
```diff
@@ -62,13 +134,13 @@ The `signOut` method will remove the current session.
...
 
-Call `getProvider` to get the Ethereum provider instance.
+Call `getProvider()` to get the Ethereum provider instance.
 
 ```typescript
 safeAuthKit.getProvider();
 ```
 
-We expose two methods for listening to events. In the case of the `Web3AuthAdapter` we can listening to all the events listed [here](https://web3auth.io/docs/sdk/web/modal/initialize#subscribing-the-lifecycle-events).
+We expose two methods for listening to events. In the case of the `Web3AuthAdapter` we can listen to all the events listed [here](https://web3auth.io/docs/sdk/web/modal/initialize#subscribing-the-lifecycle-events).
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:143]:
```suggestion
We expose two methods for listening to events. In the case of the `Web3AuthModalPack` we can listen to all the events listed [here](https://web3auth.io/docs/sdk/web/modal/initialize#subscribing-the-lifecycle-events).
```

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 8]
```diff
@@ -4,51 +4,123 @@ The [Auth kit](https://github.com/safe-global/safe-core-sdk/tree/main/packages/a
...
 ### Install dependencies
 
 ```bash
 yarn add @safe-global/auth-kit @web3auth/base @web3auth/modal @web3auth/openlogin-adapter
 ```
 
-### How to use
+If you run `yarn start` and see an error similar to `Module not found: Error: Can't resolve 'crypto'`, see Web3Auth's [Webpack 5 Polyfills Issue](https://web3auth.io/docs/troubleshooting/webpack-issues) for more context on how to fix or [commit 85ccd22](https://github.com/5afe/safe-space/pull/12/commits/85ccd22b8528d7eacd013a8e3f5bb0093c85b081) for the necessary code changes to fix the Polyfill issues.
+
+Note: You might also need to also install `browserify-zlib` (`yarn add browserify-zlib`), which is not included in the Web3Auth documentation.
```
Comment (yagopv) [inline learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:35]: Out of curiosity, I had never installed this to solve this issue. Is this specific to create-react-app and Web3Auth? I’m using vite and a plugin to address this known issue.

Comment (ademidun) [reply learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:35]: Yes, this is a troubleshooting step. Some users may face this issue some may not face this issue. I faced this issue. Which is why I say "**if** you run `yarn start` and see an error"

File: learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md [hunk 9]
```diff
@@ -92,9 +164,42 @@ const safeAuthKit = await SafeAuthKit.init(web3AuthAdapter, {
...
-Once you are logged in using the Auth kit you can get the associated web3 provider using the `safeAuthKit.getProvider()` method. If you want to make any kind of operation afterward, you can use this method with your favorite web3 library:
+The Auth Kit can be combined with the [Protocol Kit](./protocol-kit/) to connect to a Safe using the `provider` and `signer` of the currently authenticated account.
+
+Once connected, you can use any of the methods available in the [Safe Core SDK](https://github.com/safe-global/safe-core-sdk/tree/main/packages/protocol-kit#sdk-api).
+
+```typescript
+import { ethers } from 'ethers'
+import { EthersAdapter } from '@safe-global/protocol-kit'
+
+provider = new ethers.providers.Web3Provider(safeAuthKit.getProvider()!);
```
[the remark this answers is not in the document]
Comment (ademidun) [reply learn/safe-core/safe-core-account-abstraction-sdk/auth-kit.md:177]:
```suggestion
provider = new ethers.providers.Web3Provider(safeAuthKit.getProvider());
```

**Review by yagopv — APPROVED** [review]

**Pull request merged**